Assets/Scripts/ProceduralGeneration.cs

```diff
@@ -183,36 +185,38 @@ private void GenerateChunk(Vector3Int chunkStart)
 
     private void GenerateChunkMedicalCanisters(Vector3 unitCubeCenter)
     {
-        var placementVariationRange = (unitCubeSize - canisterRadius) / 2f;
-        var placementVariation = new Vector3(
-            Random.Range(-placementVariationRange, placementVariationRange),
-            Random.Range(-placementVariationRange, placementVariationRange),
-            Random.Range(-placementVariationRange, placementVariationRange)
-        );
-
-        var canisterPosition = unitCubeCenter + placementVariation;
-        var canister = Instantiate(medicalCanisterPrefab, canisterPosition, Quaternion.identity, medicalCanisterCollection.transform);
-        canister.SetActive(false);
-        AddEntity(canister);
+        var canisterPosition = RandomEntityPosition(unitCubeCenter, canisterRadius);
+        PlaceEntity(canisterPosition, medicalCanisterPrefab, medicalCanisterCollection.transform);
     }
     
     private void GenerateChunkAsteroids(Vector3 unitCubeCenter)
     {
         if (Random.value > asteroidDensity) return;
         
         var asteroidSize = Random.Range(asteroidSizeRange[0], asteroidSizeRange[1]);
-        var placementVariationRange = (unitCubeSize - asteroidSize) / 2f;
+        var asteroidPosition = RandomEntityPosition(unitCubeCenter, asteroidSize);
+        var asteroid = PlaceEntity(asteroidPosition, asteroidPrefab, asteroidsCollection.transform);
+        asteroid.transform.localScale *= asteroidSize;
```

Comment:
nit: it would look prettier if all the size stuff was in one place, which could be a one liner: `asteroid.transform.localScale *= Random.Range(asteroidSizeRange[0], asteroidSizeRange[1]);`